pkg/apiserver/archive/archive.go

```diff
@@ -472,3 +480,181 @@ func (s *Service) batchDeleteSchedule(c *gin.Context) {
 
 	c.JSON(http.StatusOK, StatusResponse{Status: "success"})
 }
+
+// @Summary Get archived workflow.
+// @Description Get archived workflow.
+// @Tags archives
+// @Produce json
+// @Param namespace query string false "namespace"
+// @Param name query string false "name"
+// @Success 200 {array} Archive
+// @Router /archives/workflows [get]
+// @Failure 500 {object} utils.APIError
+func (s *Service) listWorkflow(c *gin.Context) {
+	name := c.Query("name")
+	ns := c.Query("namespace")
+
+	metas, err := s.workflowStore.ListMeta(context.Background(), ns, name, true)
+	if err != nil {
+		c.Status(http.StatusInternalServerError)
+		_ = c.Error(utils.ErrInternalServer.NewWithNoMessage())
+		return
+	}
+
+	archives := make([]Archive, 0)
+
+	for _, meta := range metas {
+		parsedTime, err := time.Parse(time.RFC3339, meta.CreatedAt)
+		if err != nil {
+			c.Status(http.StatusInternalServerError)
+			_ = c.Error(utils.ErrInternalServer.NewWithNoMessage())
+			return
+		}
+		archives = append(archives, Archive{
+			UID:       meta.UID,
+			Kind:      v1alpha1.KindWorkflow,
+			Namespace: meta.Namespace,
+			Name:      meta.Name,
+			CreatedAt: parsedTime,
+		})
+	}
+
+	c.JSON(http.StatusOK, archives)
+}
+
+// @Summary Get the detail of an archived workflow.
+// @Description Get the detail of an archived workflow.
+// @Tags archives
+// @Produce json
+// @Param uid query string true "uid"
+// @Success 200 {object} Detail
+// @Router /archives/workflows/{uid} [get]
+// @Failure 500 {object} utils.APIError
+func (s *Service) detailWorkflow(c *gin.Context) {
+	var (
+		err    error
+		detail Detail
+	)
+	uid := c.Param("uid")
+
+	if uid == "" {
+		c.Status(http.StatusBadRequest)
+		_ = c.Error(utils.ErrInvalidRequest.New("uid cannot be empty"))
+		return
+	}
+
+	meta, err := s.workflowStore.FindByUID(context.Background(), uid)
+	if err != nil {
+		if gorm.IsRecordNotFoundError(err) {
+			c.Status(http.StatusInternalServerError)
+			_ = c.Error(utils.ErrInvalidRequest.New("the archive schedule is not found"))
+		} else {
+			c.Status(http.StatusInternalServerError)
+			_ = c.Error(utils.ErrInternalServer.NewWithNoMessage())
+		}
+		return
+	}
+
+	sch := &v1alpha1.Workflow{}
```

Comment:
Should "sch" be renamed? Change to "workflow" related?

Reply:
fixed! PTAL!